docs/madr/decisions/031-locality-aware-load-balancing.md

```diff
@@ -0,0 +1,300 @@
+
+(5) At last we should apply some fallback if no dataplanes from previous configuration are available, at this moment we allow `Fail` and `Any` fallback. Default value `Any`'
+
+### API examples based on use cases
+
+#### Keeping traffic in same zone (disable cross zone traffic)
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+```
+
+![Use case 1](assets/031/use_case_1.png)
+
+#### Load balancing based on priorities within a zone. Example: node → AZ → region
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: ["k8s.io/node", "k8s.io/az"]
+```
+
+![Use case 2](assets/031/use_case_2.png)
+![Use case 2](assets/031/use_case_2.png)
+![Use case 2](assets/031/use_case_2.png)
+
+#### Prefer traffic in same zone with fallback to any other zone
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        fallback: Any
+```
+
+![Use case 3](assets/031/use_case_3_1.png)
+![Use case 3](assets/031/use_case_3_2.png)
+
+#### Load balance equally to list of zones
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - groups:
+          - zones: ["zone-1", "zone-3"]
+        fallback: Fail
+```
+
+![Use case 4](assets/031/use_case_4_1.png)
+![Use case 4](assets/031/use_case_4_2.png)
+![Use case 4](assets/031/use_case_4_3.png)
+
+#### When leaving zone, target specific zones first
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - groups:
+          - zones: ["zone-1"]
+        - groups:
+          - zones: ["zone-3"]
+```
+
+![Use case 5](assets/031/use_case_5_1.png)
+![Use case 5](assets/031/use_case_5_2.png)
+![Use case 5](assets/031/use_case_5_3.png)
+
+#### Exclude single zone from load balancing
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - groups:
+          - zones: ["zone-1"]
+        fallback: Fail
+```
+
+![Use case 6](assets/031/use_case_6_1.png)
+![Use case 6](assets/031/use_case_6_2.png)
+![Use case 6](assets/031/use_case_6_3.png)
+
+TODO What if i have lots of zones and don’t want to write all of them during the incident. I would like to just exclude single zone, do we want to add something like:
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - exclude:
+            zones: ["zone-3"]
```

Comment:
Does it make sense to exclude zone at a level? Isn't the exclude just like `fallback` ? 

So we'd have:

```
crossZone:
  groups:
     - zones: ["zone-1", "zone-2"]
   fallback:
      type: Any
      exclude:
         zones: ["zone-4"]
```

Reply:
I've changed the model. 

```
crossZone:
  - groups:
     - name: eu
       zones: ["zone-1", "zone-2"]
     - name: us
       zones: ["zone-3"]
  - groups:
    - name: eu
      zones: ["zone-5"]
   fallback:
      type: Any
      exclude:
         zones: ["zone-4"]
```

this model allows to match groups in case of more than one priority for each group